Triggers with entity id template

r-t-s opened the pull request:
## Proposed change
<!-- 
    Describe the big picture of your changes here to communicate to the
    maintainers why we should accept this pull request. If it fixes a bug
    or resolves a feature request, be sure to link to that issue in the 
    additional information section.
-->
This adds the option to use a template for the entity_id property of State and Numeric State triggers.
This allows a blueprint to use the result of a selector to specify the entities for a trigger. 
Example blueprint is as follows:
```
blueprint:
  name: Test State Entity ID Template
  description: Demonstrate using a selector to select multiple entities for a State Trigger
  domain: automation
  source_url: https://github.com/home-assistant/core/blob/dev/homeassistant/components/automation/blueprints/test.yaml
  input:
    entities:
      name: Trigger Entities
      selector:
        target:
          entity:
            domain: switch

trigger_variables:
  entities: !input entities

trigger:
  platform: state
  id: StateTemplate
  entity_id: "{{entities.entity_id}}"
action:
  - service: logbook.log
    data:
      name: Triggered
      message: "Trigger:{{trigger}}"

```



## Type of change
<!--
    What types of changes does your PR introduce to our documentation/website?
    Put an `x` in the boxes that apply. You can also fill these out after
    creating the PR.
-->

- [ ] Spelling, grammar or other readability improvements (`current` branch).
- [ ] Adjusted missing or incorrect information in the current documentation (`current` branch).
- [ ] Added documentation for a new integration I'm adding to Home Assistant (`next` branch).
  - [ ] I've opened up a PR to add logo's and icons in [Brands repository](https://github.com/home-assistant/brands).
- [x] Added documentation for a new feature I'm adding to Home Assistant (`next` branch).
- [ ] Removed stale or deprecated documentation.

## Additional information
<!--
    Details are important, and help maintainers processing your PR.
    Please be sure to fill out additional details, if applicable.
-->

- Link to parent pull request in the codebase: https://github.com/home-assistant/core/pull/52851
- Link to parent pull request in the Brands repository: 
- This PR fixes or closes issue: 

## Checklist
<!--
    Put an `x` in the boxes that apply. You can also fill these out after
    creating the PR. If you're unsure about any of them, don't hesitate to ask.
    We're here to help! This is simply a reminder of what we are going to look
    for before merging your code.
-->

- x ] This PR uses the correct branch, based on one of the following:
  - I made a change to the existing documentation and used the `current` branch.
  - I made a change that is related to an upcoming version of Home Assistant and used the `next` branch.
- [x] The documentation follows the Home Assistant documentation [standards][].

[standards]: https://developers.home-assistant.io/docs/documenting/standards

dependabot[bot] committed:
Bump ruby/setup-ruby from 1.75.0 to 1.76.0 (#18424)

Bumps [ruby/setup-ruby](https://github.com/ruby/setup-ruby) from 1.75.0 to 1.76.0.
- [Release notes](https://github.com/ruby/setup-ruby/releases)
- [Commits](https://github.com/ruby/setup-ruby/compare/v1.75.0...v1.76.0)

---
updated-dependencies:
- dependency-name: ruby/setup-ruby
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

TheDK committed:
Update trigger.markdown (#18425)

Fixed typo.

DubhAd committed:
Highlighting breaking change/prerequisite (#18426)

Based off the breaking change in 2021.7 (https://github.com/home-assistant/core/pull/51628), highlighting the requirements here

RubenKelevra committed:
2021-07-07-release-20217.markdown: fix typos (#18418)

crazystress committed:
Update foscam.markdown (#18430)

frenck committed:
2021.7.1

chemelli74 committed:
Update vendor link to the international site (#18417)

chemelli74 committed:
Add Fritz missing info for device profiles and port forwards (#18431)

* Add info for device profiles and port forwards

* Update source/_integrations/fritz.markdown

Co-authored-by: Franck Nijhof <frenck@frenck.nl>

Co-authored-by: Franck Nijhof <frenck@frenck.nl>

r-t-s committed:
Add template option to entity_id for state and numeric_state triggers

r-t-s committed:
Add template option to entity_id for state and numeric_state triggers

r-t-s committed:
Triggers with entity id template

r-t-s committed:
Add template option to entity_id for state and numeric_state triggers

ludeeus committed:
Add missing ha_category (#18472)

ludeeus committed:
Inject post.url for anchor links (#18475)

dependabot[bot] committed:
Bump textlint from 12.0.0 to 12.0.2 (#18468)

Bumps [textlint](https://github.com/textlint/textlint) from 12.0.0 to 12.0.2.
- [Release notes](https://github.com/textlint/textlint/releases)
- [Changelog](https://github.com/textlint/textlint/blob/master/CHANGELOG.md)
- [Commits](https://github.com/textlint/textlint/compare/v12.0.0...v12.0.2)

---
updated-dependencies:
- dependency-name: textlint
  dependency-type: direct:development
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Bump dessant/lock-threads from 2.0.3 to 2.1.1 (#18469)

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

frenck committed:
Fix incorrect default scan interval for Luftdaten (#18476)

ludeeus committed:
Add missing tag (#18462)

frenck reviewed (changes requested):
There are unrelated commits in this PR. Please perform a git rebase of this PR branch onto the latest upstream `next` branch to clean it up.

HeimdallMidgard committed:
Update mqtt_room.markdown Added my ESP-32-BLE-Scanner for Home Assistant (#18465)

* Update mqtt_room.markdown

Added my ESP-32-BLE-Scanner so more ppl can find it :). If you want to make adjustments go for it. Feedback etc. is also nice.

* Tweak

Co-authored-by: Franck Nijhof <frenck@frenck.nl>

pfefferle committed:
fixed typo (#18452)

oxan committed:
Clarify and fix formatting for MQTT light JSON schema (#18458)

dependabot[bot] committed:
Bump rake from 13.0.4 to 13.0.6 (#18442)

Bumps [rake](https://github.com/ruby/rake) from 13.0.4 to 13.0.6.
- [Release notes](https://github.com/ruby/rake/releases)
- [Changelog](https://github.com/ruby/rake/blob/master/History.rdoc)
- [Commits](https://github.com/ruby/rake/compare/v13.0.4...v13.0.6)

---
updated-dependencies:
- dependency-name: rake
  dependency-type: direct:development
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

koying committed:
Clarifies usage of trusted_proxies (#18447)

* Clarifies usage of trusted_proxies

As discussed in https://github.com/home-assistant/core/issues/52736

* Tweak

Co-authored-by: Franck Nijhof <frenck@frenck.nl>

frenck committed:
Add missing webhook to default config docs (#18479)

jbouwh committed:
MQTT Fan docs - sorted config and small text corrections (#18454)

* sorted config and small text corrections

* remove unneeded raw tags

dependabot[bot] committed:
Bump actions/setup-node from 2.1.5 to 2.2.0 (#18361)

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

balloob committed:
2021.7.2

r-t-s committed:
Merge branch 'home-assistant:current' into triggers_with_entity_id_template

frenck commented:
![image](https://user-images.githubusercontent.com/195327/125365423-81e65700-e374-11eb-9037-b0a4a660adce.png)

That is a merge; Please git rebase, not merge.

r-t-s commented:
I knew I screwed up when I thought I was doing  a push which turned into a
merge  ... but now I can't figure out how to correct things.
I think I made things worse when I tried to backout the merge.
I think I will reset my repository to the latest version.
Create a NEW branch based on 'next' , reintegrate my changes, and create a
new PR.
I will update the code PR with a reference to the new home-assistant.io PR
I know this causes problems for your automated workflows ... I will
continue to learn how to correctly use git.



On Mon, Jul 12, 2021 at 5:52 PM Franck Nijhof ***@***.***>
wrote:

> [image: image]
> <https://user-images.githubusercontent.com/195327/125365423-81e65700-e374-11eb-9037-b0a4a660adce.png>
>
> That is a merge; Please git rebase, not merge.
>
> —
> You are receiving this because you were assigned.
> Reply to this email directly, view it on GitHub
> <https://github.com/home-assistant/home-assistant.io/pull/18455#issuecomment-878648699>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ACVV5AO2YXMJTRNW6VC3JTDTXNWZ3ANCNFSM5AEP4LWQ>
> .
>


-- 

*Richard T. Schaefer      E-mail: **schaefer@RTS-Services.com*
<schaefer@RTS-Services.com>

*RTS Services Inc            Web: **http://RTS-Services.com*
<http://rts-services.com/>



*1675 Hickory Creek Rd.   ___        Voice:  (512) 923-2755 Marble
Falls, TX 78654    |                        `---------(*)---------'*

r-t-s commented:
I think I found the source of my problems ...
I do
    git fetch upstream BRANCH
    git rebase upstream/BRANCH
    git  push origin HEAD

It prompts me to do a pull ... and I did it !!
I see from https://developers.home-assistant.io/docs/development_catching_up
that I was supposed to do a --force.
I am not sure I know how to clean things up  now.
I may have to reject all of my releases (core and home-assistant.io) and
restart.



On Mon, Jul 12, 2021 at 7:19 PM Richard T Schaefer <
***@***.***> wrote:

> I knew I screwed up when I thought I was doing  a push which turned into a
> merge  ... but now I can't figure out how to correct things.
> I think I made things worse when I tried to backout the merge.
> I think I will reset my repository to the latest version.
> Create a NEW branch based on 'next' , reintegrate my changes, and create a
> new PR.
> I will update the code PR with a reference to the new home-assistant.io PR
> I know this causes problems for your automated workflows ... I will
> continue to learn how to correctly use git.
>
>
>
> On Mon, Jul 12, 2021 at 5:52 PM Franck Nijhof ***@***.***>
> wrote:
>
>> [image: image]
>> <https://user-images.githubusercontent.com/195327/125365423-81e65700-e374-11eb-9037-b0a4a660adce.png>
>>
>> That is a merge; Please git rebase, not merge.
>>
>> —
>> You are receiving this because you were assigned.
>> Reply to this email directly, view it on GitHub
>> <https://github.com/home-assistant/home-assistant.io/pull/18455#issuecomment-878648699>,
>> or unsubscribe
>> <https://github.com/notifications/unsubscribe-auth/ACVV5AO2YXMJTRNW6VC3JTDTXNWZ3ANCNFSM5AEP4LWQ>
>> .
>>
>
>
> --
>
> *Richard T. Schaefer      E-mail: **schaefer@RTS-Services.com*
> <schaefer@RTS-Services.com>
>
> *RTS Services Inc            Web: **http://RTS-Services.com*
> <http://rts-services.com/>
>
>
>
> *1675 Hickory Creek Rd.   ___        Voice:  (512) 923-2755 Marble
> Falls, TX 78654    |                        `---------(*)---------'*
>


-- 

*Richard T. Schaefer      E-mail: **schaefer@RTS-Services.com*
<schaefer@RTS-Services.com>

*RTS Services Inc            Web: **http://RTS-Services.com*
<http://rts-services.com/>



*1675 Hickory Creek Rd.   ___        Voice:  (512) 923-2755 Marble
Falls, TX 78654    |                        `---------(*)---------'*

r-t-s commented:
I will have to re-release this ... I corrupted my branch.
